### PR TITLE
Add BigBoy macro pad

### PR DESCRIPTION
**Pull request by tominabox1**
## Description

add big boy macro pad

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x ] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [ ] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x ] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [ ] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/tominabox1/bigboy/config.h [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+/* USB Device descriptor parameter */
+#define VENDOR_ID 0x7431
+#define PRODUCT_ID 0x6262
+#define DEVICE_VER 0x0001
+#define MANUFACTURER tominabox1
+#define PRODUCT BigBoy
+
+/* key matrix size */
+#define DIRECT_PINS { \
+    { D0, B1, B0, C7 }, \
```
Comment (zvecr) [inline keyboards/tominabox1/bigboy/config.h:30]: 2x4, but layout macro is 2x3?

Comment (tominabox1) [reply keyboards/tominabox1/bigboy/config.h]: oops thanks, forgot to fix that

File: keyboards/tominabox1/bigboy/config.h [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+#define UNUSED_PINS
+
+#define ENCODERS_PAD_A { C6 }
+#define ENCODERS_PAD_B { B6 }
+#define ENCODER_RESOLUTION 2
+
+#define RGB_DI_PIN B5
+
+#ifdef RGB_DI_PIN
+#define RGBLIGHT_ANIMATIONS
```
Comment (drashna) [inline keyboards/tominabox1/bigboy/config.h:43]: Could you explicitly define which animations are enabled? 
```suggestion
#define RGBLIGHT_EFFECT_BREATHING
#define RGBLIGHT_EFFECT_RAINBOW_MOOD
#define RGBLIGHT_EFFECT_RAINBOW_SWIRL
#define RGBLIGHT_EFFECT_SNAKE
#define RGBLIGHT_EFFECT_KNIGHT
#define RGBLIGHT_EFFECT_CHRISTMAS
#define RGBLIGHT_EFFECT_STATIC_GRADIENT
#define RGBLIGHT_EFFECT_RGB_TEST
#define RGBLIGHT_EFFECT_ALTERNATING
#define RGBLIGHT_EFFECT_TWINKLE
```

File: keyboards/tominabox1/bigboy/keymaps/default/keymap.c [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+bool encoder_update_user(uint8_t index, bool clockwise) {
+  if (index == 0) { /* First encoder */
+    if (clockwise) {
+      tap_code(KC_VOLU);
+    } else {
+      tap_code(KC_VOLD);
+    }
+  }
+  return false;
+}
```
Comment (drashna) [inline keyboards/tominabox1/bigboy/keymaps/default/keymap.c:45]:
```suggestion
```

File: keyboards/tominabox1/bigboy/bigboy.c [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include "bigboy.h"
```
Comment (drashna) [inline keyboards/tominabox1/bigboy/bigboy.c:17]: This allows for a default behavior to be defined, without the need for the user to specify the behavior.  This is very useful for the configurator which doesn't have a way to configure this behavior, currently.

```suggestion
#include "bigboy.h"

#ifdef ENCODER_ENABLE
bool encoder_update_kb(uint8_t index, bool clockwise) {
    if (!encoder_update_user(index, clockwise)) { return false; }
    if (index == 0) {
        if (clockwise) {
            tap_code_delay(KC_VOLU, 10);
        } else {
            tap_code_delay(KC_VOLD, 10);
        }
    }
    return true;
}
#endif
```

**Review by drashna — CHANGES_REQUESTED** [review]

File: keyboards/tominabox1/bigboy/readme.md [hunk 5]
```diff
@@ -0,0 +1,20 @@
+# BigBoy
+
+A macropad designed by dingusxmcgee_ and tominabox1.
+
+![bigboy](update .jpg)
```
Comment (drashna) [inline keyboards/tominabox1/bigboy/readme.md:5]: image?

**Review by drashna — APPROVED** [review]

**Comment by tominabox1** [comment]
any update on this

**Review by tzarc — APPROVED** [review]

**Pull request merged**